### ISSUE TITLE
REST API to return Transaction Binding 

### ISSUE THREAD

**Issue by mgamanho** [hyperledger-archives/fabric#1775]
## Description

Hello, would it make sense to return the Transaction Binding for a given enrollment ID, from a crypto perspective? In other words, is it missing from the REST API on purpose? The idea would be to have sign/verify functions purely REST based.

**Comment by christo4ferris** [comment]
@smithbk can you please respond to this? Thanks

**Comment by smithbk** [comment]
@mgamanho Yes, we could enhance the REST API to require signatures/etc in order to make it secure.  I assume this is what you're asking.  However, we've been focusing on building a node client SDK and chose to use GRPC because it is more efficient.  We hope other language SDKs will follow soon (e.g. java).  So if you use a client SDK which uses GRPC under the covers, then it should be easier to use and supporting crypto via REST becomes less important.   That said, given the amount of time it takes to develop client SDKs in various languages, making REST secure may be something to reconsider.
What language are you using for your client?

**Comment by zmanian** [comment]
REST API calls seem to escape both the T-Cert mechanism for rate limiting transactions and limit the effectiveness of the client registration by turning usernames into a security token. I suspect any solution should address both issues.

**Comment by mgamanho** [comment]
Hi, the client language is javascript. I've actually changed the code to do what I needed in the meantime, i.e.: return a signature. IMO gRPC might limit adoption somewhat.

Another issue I saw was, as @zmanian points out, that T-Certs are not quite well integrated with REST, essentially unusable as nothing ties one call to another in the same trans., so that would need looking at, probably first I would say.
